samples/msal-node-samples/standalone-samples/b2c-auth-code-pkce/README.md

```diff
@@ -0,0 +1,248 @@
+
+### Registration
+
+This sample comes with a pre-registered application for demo purposes. If you would like to use your own **Azure AD B2C** tenant and application, follow the steps below:
+
+1. [Create an Azure Active Directory B2C tenant](https://docs.microsoft.com/azure/active-directory-b2c/tutorial-create-tenant)
+2. [Register a web application in Azure Active Directory B2C](https://docs.microsoft.com/azure/active-directory-b2c/tutorial-register-applications?tabs=app-reg-ga)
+3. [Create user flows in Azure Active Directory B2C](https://docs.microsoft.com/azure/active-directory-b2c/tutorial-create-user-flows)
+
+### Configuration
+
+In `policies.js`, we create a `b2cPolicies` object to store authority strings for initiating each user-flow:
+
+```javascript
+const b2cPolicies = {
+    authorities: {
+        signUpSignIn: {
+            authority: "https://fabrikamb2c.b2clogin.com/fabrikamb2c.onmicrosoft.com/B2C_1_susi",
+        },
+        resetPassword: {
+            authority: "https://fabrikamb2c.b2clogin.com/fabrikamb2c.onmicrosoft.com/B2C_1_reset",
+        },
+        editProfile: {
+            authority: "https://fabrikamb2c.b2clogin.com/fabrikamb2c.onmicrosoft.com/B2C_1_edit_profile"
+        }
+    },
+    authorityDomain: "fabrikamb2c.b2clogin.com"
+}
+```
+
+In `index.js`, we setup the configuration object expected by MSAL Node `publicClientApplication` class constructor:
+
+```javascript
+const publicClientConfig = {
+    auth: {
+        clientId: "e6e1bea3-d98f-4850-ba28-e80ed613cc72",
+        authority: policies.authorities.signUpSignIn.authority, //signUpSignIn policy is our default authority
+        knownAuthorities: [policies.authorityDomain], // mark your tenant's custom domain as a trusted authority
+        redirectUri: "http://localhost:3000/redirect",
+    },
+    system: {
+        loggerOptions: {
+            loggerCallback(loglevel, message, containsPii) {
+                console.log(message);
+            },
+            piiLoggingEnabled: false,
+            logLevel: msal.LogLevel.Verbose,
+        }
+    }
+};
+```
+
+MSAL enables PKCE in the Authorization Code Grant Flow by including the `codeChallenge` and `codeChallengeMethod` parameters in the request passed into `getAuthCodeUrl()` API, as well as the `codeVerifier` parameter in the second leg (`acquireTokenByCode()` API).
+
+Generating the `codeVerifier` and the `codeChallenge` is the client application's responsibility. For this sample, you can either implement your own PKCE code generation logic or use an existing tool to manually generate a **Code Verifier** and **Code Challenge**, plugging them into the `pkceCodes` object below.
+
+For details on implementing your own PKCE code generation logic, consult the PKCE specification `https://tools.ietf.org/html/rfc7636#section-4`
```

Comment:
@samuelkubai updated that our PKCE helpers are indeed functional in node. Can @hectormmg / @derisen please confirm the same for `msal-node` in AAD and B2C samples? We can update the docs once we are confident.

Reply:
We found out that there's a bug that's causing some inconsistency with the result of the PKCE helpers that is to be fixed with this [pull request ](https://github.com/AzureAD/microsoft-authentication-library-for-js/pull/2687)

Reply:
@sameerag I have added an instruction to use our `CryptoProvider` to generate the PKCE codes.

Reply:
Thanks @samuelkubai. Looks good.